client/js/components/App.react.jsx

```diff
@@ -39,6 +73,37 @@ export default class App extends React.Component {
             .tap((data) => this.setState({trailGeo: data}));
     }
 
+    setUserLocation(userLocation) {
+        var parksWithDistance = this.state.allParks.map((park) => {
+            var toPoint = turf.point([userLocation[1], userLocation[0]]);
+            var fromPoint = turf.point([park.center[1], park.center[0]]);
+
+            park.distance = turf.distance(toPoint, fromPoint, 'miles');
+
+            return park;
+        });
+
+        this.setState({
+            allParks: parksWithDistance,
+            userLocation: userLocation
+        });
+    }
+
+    applyFilters(filter) {
+        var visibleParks = this.state.allParks.filter((park) => {
+            // FIXME: Convert park ids to numbers when we generate the data
+            var matchingAmenity = !!this.state.amenityLookup && this.state.amenityLookup[filter] && this.state.amenityLookup[filter].map(Number).indexOf(park.park_id) !== -1;
```

Comment:
yuckaroo!1!!

A helper function maybe?

```javascript
function safeLookup(obj, filter) {
  return !!obj && _.contains(obj[filter].map(Number), park.park_id);
}
```
Or whatever the appropriate lo-dash is